pandas/sparse/frame.py

```diff
@@ -187,9 +215,84 @@ def _init_matrix(self, data, index, columns, dtype=None):
         if len(index) != N:
             raise ValueError('Index length mismatch: %d vs. %d' %
                              (len(index), N))
+        return index, columns
 
-        data = dict([(idx, data[:, i]) for i, idx in enumerate(columns)])
-        return self._init_dict(data, index, columns, dtype)
+    def as_matrix(self, columns=None, sparse=False):
```

Comment:
I understand why you are changing / adding this. BUT this is actually a major API change. This is what ``.values`` calls and as a result the return types are: dense numpy arrays / pandas extension types (categorical).

Not against this (in fact I think its a good idea), but split this out into a separate issue. Then the impl would actually be done in the block manager.